source/numerics/data_out_resample.cc

```diff
@@ -0,0 +1,192 @@
+{}
+
+
+
+template <int dim, int patch_dim, int spacedim>
+void
+DataOutResample<dim, patch_dim, spacedim>::update_mapping(
+  const Mapping<dim, spacedim> &mapping,
+  const unsigned int            n_subdivisions)
+{
+  this->mapping = &mapping;
+  this->point_to_local_vector_indices.clear();
+
+  FE_Q_iso_Q1<patch_dim, spacedim> fe(
+    std::max<unsigned int>(1, n_subdivisions));
+  patch_dof_handler.distribute_dofs(fe);
+
+  std::vector<std::pair<types::global_dof_index, Point<spacedim>>> points_all;
+
+  Quadrature<patch_dim> quadrature(fe.get_unit_support_points());
+
+  FEValues<patch_dim, spacedim> fe_values(*patch_mapping,
+                                          fe,
+                                          quadrature,
+                                          update_quadrature_points);
+
+  std::vector<types::global_dof_index> dof_indices(fe.n_dofs_per_cell());
+
+  IndexSet active_dofs;
+  DoFTools::extract_locally_active_dofs(patch_dof_handler, active_dofs);
+  partitioner = std::make_shared<Utilities::MPI::Partitioner>(
+    patch_dof_handler.locally_owned_dofs(), active_dofs, MPI_COMM_WORLD);
+
+  for (const auto &cell : patch_dof_handler.active_cell_iterators())
+    {
+      if (cell->is_locally_owned() == false)
+        continue;
+
+      fe_values.reinit(cell);
+
+      cell->get_dof_indices(dof_indices);
+
+      for (unsigned int i = 0; i < dof_indices.size(); ++i)
+        points_all.emplace_back(partitioner->global_to_local(dof_indices[i]),
+                                fe_values.quadrature_point(i));
+    }
+
+  std::sort(points_all.begin(),
+            points_all.end(),
+            [](const auto &a, const auto &b) { return a.first < b.first; });
+  points_all.erase(std::unique(points_all.begin(),
+                               points_all.end(),
+                               [](const auto &a, const auto &b) {
+                                 return a.first == b.first;
+                               }),
+                   points_all.end());
+
+  std::vector<Point<spacedim>> points;
+
+  for (const auto &i : points_all)
+    {
+      point_to_local_vector_indices.push_back(i.first);
+      points.push_back(i.second);
+    }
+
+  rpe.reinit(points, *this->triangulation, *this->mapping);
+}
+
+
+
+template <int dim, int patch_dim, int spacedim>
+void
+DataOutResample<dim, patch_dim, spacedim>::build_patches(
+  const Mapping<dim, spacedim> &                                mapping,
+  const unsigned int                                            n_subdivisions,
+  const typename DataOut<patch_dim, spacedim>::CurvedCellRegion curved_region)
+{
+  this->update_mapping(mapping, n_subdivisions);
+  this->build_patches(curved_region);
+}
+
+
+
+template <int dim, int patch_dim, int spacedim>
+void
+DataOutResample<dim, patch_dim, spacedim>::build_patches(
+  const typename DataOut<patch_dim, spacedim>::CurvedCellRegion curved_region)
+{
+  patch_data_out.clear();
+
+  if (rpe.is_ready() == false)
```

Comment:
Does rpe attach to signals that invalidate data whenever the triangulation changes?

Reply:
Yes.